cmd/terramate/cli/run.go

```diff
@@ -421,3 +434,64 @@ func (c *cli) loadAllStackEnvs(runs []runContext) (map[prj.Path]runutil.EnvVars,
 	}
 	return stackEnvs, nil
 }
+
+func (c *cli) createCloudPreview(runs []runContext) map[string]string {
+	previewRuns := make([]previews.RunContext, len(runs))
+	for i, run := range runs {
+		previewRuns[i] = previews.RunContext{
+			Stack: run.Stack,
+			Cmd:   run.Cmd,
+		}
+	}
+
+	affectedStacks, err := c.computeSelectedStacks(false, true, cloudstack.NoFilter)
+	if err != nil {
+		fatal("computing affected stacks", err)
+	}
+
+	affectedStacksMap := map[string]*config.Stack{}
+	for _, st := range affectedStacks {
+		affectedStacksMap[st.Stack.ID] = st.Stack
+	}
+
+	githubEventPath, ok := os.LookupEnv("GITHUB_EVENT_PATH")
+	if !ok {
+		fatal("env var GITHUB_EVENT_PATH not found, not generating previews", nil)
```

Comment:
The `sync` features must never abort without executing the provided `run` or `script` command. It must warn and just disable the feature. Check the `c.disableCloudFeatures(err)`.

Reply:
In this case disabling does not make sense because `--cloud-sync-preview` feature requires a timestamp to create a preview.
However, we could handle that case gracefully by using `time.Now()` and removing the `fatal`, what do you think?